packages/core/test/invoke.test.ts

```diff
@@ -2905,6 +2905,41 @@ describe('invoke', () => {
       done();
     }, 0);
   });
+
+  // https://github.com/statelyai/xstate/issues/3072
+  it('root invocations should not restart on external transitions', (done) => {
```

Comment:
We should also add a separate test that would test, in a more explicit way, that the root state is never being reentered here. That `invoke`'s implementation is piggybacking on entry/exit actions is an implementation detail so I don't think that we should assume that this is also testing that more "basic" scenario.